src/components/InputProblem.tsx

```diff
@@ -1,7 +1,92 @@
+import React, { useState, useCallback } from 'react'
 import { BaseProblemProps } from './ProblemSetBlock'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+import { mathifyElement } from '../lib/math'
+import * as Yup from 'yup'
 
-interface InputProblemProps extends BaseProblemProps {}
+interface InputProblemProps extends BaseProblemProps {
+  comparator: string
+}
+
+interface InputSchema {
+  response: string | number
+}
+
+interface InputFormValues {
+  response: string
+}
+
+export const InputProblem = ({
+  solvedCallback, exhaustedCallback, allowedRetryCallback,
+  solution, retryLimit, content, comparator, encourageResponse, buttonText, correctResponse
+}: InputProblemProps): JSX.Element => {
+  const [retriesAllowed, setRetriesAllowed] = useState(0)
+  const [inputDisabled, setInputDisabled] = useState(false)
+  const [feedback, setFeedback] = useState('')
+
+  const schema = (): Yup.SchemaOf<InputSchema> => {
+    if (comparator === 'integer') {
+      return Yup.object({
+        response: Yup.number().integer().typeError('Please provide an Integer').required('Please provide an Integer')
+      })
+    }
+    if (comparator === 'float') {
+      return Yup.object({
+        response: Yup.number().typeError('Please provide an number').required('Please provide a number')
+      })
+    }
+    return Yup.object({
+      response: Yup.string().trim().required('Please provide valid input')
+    })
+  }
+
+  const contentRefCallback = useCallback((node: HTMLDivElement | null): void => {
+    if (node != null) {
+      mathifyElement(node)
+    }
+  }, [])
+
+  const handleSubmit = async (values: InputFormValues): Promise<void> => {
+    if (values.response.trim() === solution.trim()) {
+      setFeedback(correctResponse)
+      solvedCallback()
+      setInputDisabled(true)
+    } else if (retryLimit === 0 || retriesAllowed !== retryLimit) {
+      setRetriesAllowed(currRetries => currRetries + 1)
+      setFeedback(encourageResponse)
+      allowedRetryCallback()
+    } else {
+      exhaustedCallback()
+      setFeedback('No more attempts allowed')
+      setInputDisabled(true)
+    }
+  }
+  const clearFeedback = (): void => {
+    setFeedback('')
+  }
+  return (
+  <div className="os-raise-bootstrap">
+
+    <div className="my-3" ref={contentRefCallback} dangerouslySetInnerHTML={{ __html: content }} />
+    <Formik
+          initialValues={{ response: '' }}
+          onSubmit={handleSubmit}
+          validationSchema={schema}
+        >
+          {({ isSubmitting, setFieldValue }) => (
+            <Form >
+              <Field
+              name="response"
+              as="textarea"
+              disabled={inputDisabled}
+              onChange={(e: React.ChangeEvent<HTMLTextAreaElement>) => { clearFeedback(); setFieldValue('response', e.target.value) }}
+              className="form-control my-3" />
```

Comment:
If you wanted to spruce up the styling a bit so the input is not `full-width` per [here](https://github.com/twbs/bootstrap/blob/416f4905f13c2ca3096343e0a5a4934c8c0ced3f/scss/forms/_form-control.scss#L7), you can define / use a `.os-form-control` class in our `bootstrap.scss` as:

```css
  .os-form-control {
    /* stylelint-disable-next-line scss/at-extend-no-missing-placeholder */
    @extend .form-control;

    width: unset;
  }
```

Reply:
@P-Gill97 - Everything looks good to me on this PR except for 2 UI related items:

1. Can we add the style definition for `.os-form-control` in my previous comment [after this line](https://github.com/openstax/k12-apps-raise/blob/6c7c3d02a0e25fb0e6800800c87c737389a5d300/src/styles/bootstrap.scss#L83) in `bootstrap.scss`, and then apply the class here so it's `className="os-form-control my-3"`? That will give us the nicer Bootstrap styling on the input box versus the default browser.

2. Can we move the `ErrorMessage` on top of the `<button>` to be consistent with the other components? So it's something like:

```javascript
<ErrorMessage className="text-danger mb-3" component="div" name="response" />
<button type="submit" disabled={inputDisabled || isSubmitting} className="btn btn-outline-primary ml-3">{buttonText}</button>
```

Reply:
@P-Gill97 - Just saw your last commit and while the styling looks good now I think the element ordering is still not quite right as now the feedback is above the button. We want to be consistent with other components such as [here](https://github.com/openstax/k12-apps-raise/blob/6c7c3d02a0e25fb0e6800800c87c737389a5d300/src/components/MultiselectProblem.tsx#L110-L112) where it's:

```javascript
<ErrorMessage />
<button />
{feedback }
```

Unfortunately this consistency is by convention right now so we have to keep an eye out for it manually. I already have it on my list to create an issue to refactor some of this problem type code which will be an opportunity to codify the conventions in a single location.

Reply:
Sounds good. Fixed it.